server/container/guice/guice-common/src/test/java/org/apache/james/JamesParametersResolver.java

```diff
@@ -7,10 +7,40 @@
 
 class JamesParametersResolver implements ParameterResolver {
 
+    static class Builder {
+        @FunctionalInterface
+        interface RequireJamesServer {
+            RequireRegistrableExtension jamesServer(GuiceJamesServer jamesServer);
+        }
+
+        @FunctionalInterface
+        interface RequireRegistrableExtension {
+            ReadyToBuild registrableExtension(RegistrableExtension registrableExtension);
+        }
+
+        static class ReadyToBuild {
+            private final GuiceJamesServer jamesServer;
+            private final RegistrableExtension registrableExtension;
+
+            ReadyToBuild(GuiceJamesServer jamesServer, RegistrableExtension registrableExtension) {
+                this.jamesServer = jamesServer;
+                this.registrableExtension = registrableExtension;
+            }
+
+            JamesParametersResolver build() {
```

Comment:
when you can't do anything else than build in the last step, then you should not require this step.
`registrableExtension` should return `JamesParametersResolver` directly (and yes, it's not a builder anymore)